psycopg/psycopg/connection.py

```diff
@@ -95,27 +95,44 @@ def connect(
         timeout = timeout_from_conninfo(params)
         rv = None
         attempts = conninfo_attempts(params)
+        connection_errors: list[tuple[e._NoTracebackException, str]] = []
         for attempt in attempts:
             try:
                 conninfo = make_conninfo("", **attempt)
                 gen = cls._connect_gen(conninfo, timeout=timeout)
                 rv = waiting.wait_conn(gen, interval=_WAIT_INTERVAL)
             except e._NO_TRACEBACK as ex:
-                if len(attempts) > 1:
-                    logger.debug(
-                        "connection attempt failed: host: %r port: %r, hostaddr %r: %s",
-                        attempt.get("host"),
-                        attempt.get("port"),
-                        attempt.get("hostaddr"),
-                        str(ex),
-                    )
-                last_ex = ex
+                attempt_details = "host: {}, port: {}, hostaddr: {}".format(
+                    repr(attempt.get("host")),
+                    repr(attempt.get("port")),
+                    repr(attempt.get("hostaddr")),
```

Comment:
Is this the most useful order in which to show these 3 details or you would rather see them in a different error to spot problems more easily (hostaddr first maybe?)

Reply:
I'm not sure - my thought was there may be multiple hosts, and then multiple addresses associated with each host. I tried to do it in the order of what I thought was likely to have the least variations first, moving to the most variations on the right.

Reply:
Ok, let's leave it this way.